os/arch/arm/src/common/up_createstack.c

```diff
@@ -243,6 +252,18 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 		heapinfo_exclude_stacksize(tcb->stack_alloc_ptr);
 #endif
 		board_led_on(LED_STACKCREATED);
+#ifdef CONFIG_MPU_STACKGUARD
+		/* The smallest size that can be programmed for an MPU region is 32 bytes */
+
+		tcb->stack_guard =	(uint32_t)tcb->stack_alloc_ptr + 0x40;
+		tcb->stack_guard = ((uint32_t)tcb->stack_guard & MPU_RBAR_ADDR_MASK);
+		tcb->guard_size = ((uint32_t)tcb->stack_guard - (uint32_t)tcb->stack_alloc_ptr);
```

Comment:
The guard size will not change based on the guard start address. It must always be equal to 32 bytes.
So, instead of using a variable like guard_size, we can use a MACRO to give the size.

Reply:
We need to take care of 32 byte aligned address as well as 32 byte size for the region. 
The aligned address will result into size from 32 byte to max 64 byte.

Reply:
Right. Size must always be 32 bytes. 
But, according to above code, size will be `((uint32_t)tcb->stack_guard - (uint32_t)tcb->stack_alloc_ptr)` which is not always 32 bytes.

Reply:
yes size will always be >= 32, to make both address and size aligned to 32 byte

Reply:
calculation is changed after adding kumm_memalign